applications/app/services/TodaysNewspaperQuery.scala

```diff
@@ -0,0 +1,105 @@
+import org.joda.time.{DateTime, DateTimeZone}
+import slices.{ContainerDefinition, Fixed, FixedContainers, TTT}
+
+import scala.concurrent.Future
+
+case class TagWithContent(tag: Tag, content: ApiContent)
+case class BookSectionContent(tag: Tag, content: List[ApiContent])
+
+object TodaysNewspaperQuery extends ExecutionContexts with Dates with Logging {
+
+  def fetchTodaysPaper: Future[List[FaciaContainer]] = {
+    val today = DateTime.now(DateTimeZone.UTC)
+
+    val item = LiveContentApi.item("theguardian/mainsection")
+      .useDate("newspaper-edition")
+      .showFields("all")
+      .showElements("all")
+      .showTags("newspaper-book-section")
+      .pageSize(200)
+      .fromDate(today.withTimeAtStartOfDay())
+      .toDate(today)
+
+    LiveContentApi.getResponse(item).map { resp =>
+
+      //filter out the first page results to make a Front Page container
+      val (firstPageContent, otherContent) = resp.results.partition(content => getNewspaperPageNumber(content).contains(1))
+
+      val firstPageContainer = {
+        val content = firstPageContent.map(c => FaciaContentConvert.frontendContentToFaciaContent(Content(c)))
+        bookSectionContainer(None, Some("Front Page"), content, 0)
+      }
+
+      val unorderedBookSections = createBookSections(otherContent)
+      val orderedBookSections = orderByPageNumber(unorderedBookSections)
+
+      val bookSectionContainers = orderedBookSections.map { list =>
+        val content = list.content.map(c => FaciaContentConvert.frontendContentToFaciaContent(Content(c)))
+        bookSectionContainer(Some(list.tag.id), Some(list.tag.webTitle), content, orderedBookSections.indexOf(list) + 1)
+      }
+
+      firstPageContainer :: bookSectionContainers
+    }
+  }
+
+
+
+  private def createBookSections(contentList: List[ApiContent]): List[BookSectionContent] = {
+    val tagWithContent: List[TagWithContent] = contentList.flatMap { content =>
+      content.tags.find(_.`type` == "newspaper-book-section").map(t => TagWithContent(t, content))
+    }
+
+    //group content by booksection tag type
+    tagWithContent.groupBy(_.tag).map( bookSectionContent => BookSectionContent(bookSectionContent._1, bookSectionContent._2.map(_.content))).toList
+  }
+
+  private def orderByPageNumber(unorderedBookSections: List[BookSectionContent]): List[BookSectionContent] = {
+
+    //order content for each book section
+    val orderedContentForBookSection: List[BookSectionContent] = unorderedBookSections.map { bookSection =>
+      bookSection.copy(content = orderContentByPageNumber(bookSection.content))
+    }
+
+    //order booksections by first content item in each booksection
+    val pageNumberToFaciaContainer: List[(Int, BookSectionContent)] = orderedContentForBookSection.flatMap { bookSection =>
+      val pageNumberOpt = bookSection.content.headOption.flatMap(content => getNewspaperPageNumber(content))
+      pageNumberOpt.map(_ -> bookSection)
```

Comment:
I had similar approach for grouping the tag and content (which is a stepping stone to creating the BookSectionContent). I felt I was creating case classes for just one line. However if it it makes it more readable then it's I guess that is better. Will give it a go here.